File: Erlang/Questao08.erl
% Define o módulo do arquivo
-module(Questao08).

% Funções que serão exportadas
-export([potencia/1]).

% Definição da função potência
potencia(lista) ->
    % Tamanho da lista
    tamanho = length(lista)
    % Quantidade de subconjuntos
    maximo = trunc(math:pow(2, tamanho))
    % Gera todas as possibilidades de subconjuntos usando bitwise
    [[lists:nth(posicao + 1, lista) || posicao <- lists:seq(0, tamanho - 1), I band (1 bsl posicao) =/= 0 || I <- lists:seq(0, maximo - 1)].

File: Haskell/Questao08.hs
-- Definição da função potência
potencia :: [a] -> [[a]]
potencia [] = [[]]
potencia (x: xs) = map (x:) (potencia xs) ++ potencia xs

main :: IO ()
main = do
    let lista = -- Inserir lista no formato [a, b, c, ...]
    
    -- Imprime o conjunto potência da lista
    print (potencia lista)